contextvars.py

```diff
@@ -57,6 +57,18 @@ def default(self) -> T:
 
     # Methods that take the current context into account
 
+    def get_stack(self) -> List[Optional[T]]:
+        ec: Optional['ExecutionContext'] = get_EC()
+        values: List[Optional[T]] = []
+        while ec is not None:
+            if self in ec.lc:
+                value: T = ec.lc[self]
+                values.append(value)
+            else:
+                values.append(None)
```

Comment:
Why insert these Nones? Does the use case require maintaining parallel mappings between the EC and some data structure Nathaniel has built up?

Reply:
For Nathaniel's use case, Nones are needed to check if the topmost LC has any Trio cancel scopes/timeouts in it.  The same could be achieved if we added a new method to check if the topmost LC has a value for a `ContextVar`, but I figured that we can have just one method for advanced users that covers all such use cases.

Here's an example of a context manager that would use this method (the example uses synchronous generators, but the same idea will work for asynchronous ones):

```python

cms = contextvars.ContextVar()

class CM:
    def __init__(self, name):
        self.name = name

    def __repr__(self):
       return f'CM({self.name})'

    def __enter__(self):
        stack = cms.get_stack()
        if stack[-1] is None:
            # first 'with CM' in this LC
            cms.set([self])
        else:
            cms.get().append(self)

    def __exit__(self, *e):
        cms.get().pop()
```

Now let's use it:

```python
def gen1():
    with CM('gen1'):
        yield cms.get_stack()
        yield cms.get_stack()

def gen2():
    with CM('gen2'):
        yield next(g1)

g1 = gen1()
g2 = gen2()

with CM('topmost'):
    print('first yield:', next(g1))

    with CM('topmost2'):
        print('second yield:', next(g2))
```

The script will output:

```
first yield: [ [CM(topmost)], [CM(gen1)] ]
second yield: [ [CM(topmost),CM(topmost2)], [CM(gen2)], [CM(gen1)] ]
```

So in Trio, `ContextVar.get_stack()` will be used for the following:

1. Check if the topmost LC has a list of Trio scopes (need Nones for this).
2. Capture the entire stack of scopes in the Trio task machinery whenever we yield control to it.
3. When a new Trio task is spawned, get the stack of scopes right before calling `get_execution_context()` to manually squash Trio scopes.